docs/content/latest/develop/query-1-performance/explain-analyze.md

```diff
@@ -0,0 +1,96 @@
+| VERBOSE | boolean                         | Displays detailed information about the query plan. <br/>The default value is FALSE. |
+| COSTS   | boolean                         | Provides the estimated initial and total costs of each plan node. In addition, estimates the number of rows and the width of each row in the query plan.<br/>The default value is TRUE. |
+| BUFFERS | boolean                         | Provides information about the most input-output intensive parts of the query. <br/>The default value is FALSE. <br/>You can only use this option when ANALYZE is set to TRUE. |
+| TIMING  | boolean                         | Provides information about the actual startup time and the time spent in each node of the output. <br/>The default value is TRUE. <br/>You can only use this option when ANALYZE is set to TRUE. |
+| SUMMARY | boolean                         | Provides additional information, such as the total time after the query plan. The value of this option is TRUE when ANALYZE is set to TRUE. |
+| FORMAT  | { TEXT \| XML \| JSON \| YAML } | Allows you to define the query plan output format. <br/>The default value is TEXT. |
+
+### Examples
+
+Typically, you start by creating a table in YugabyteDB and inserting rows into it. 
+
+To create a table called `employees`, execute the following:
+
+```sql
+yugabyte=# CREATE TABLE employees(k1 int, k2 int, v1 int, v2 text, PRIMARY KEY (k1, k2));
+```
+
+To insert table rows, execute the following:
+
+```sql
+yugabyte=# INSERT INTO employees(k1, k2, v1, v2) VALUES (1, 2.0, 3, 'a'), (2, 3.0, 4, 'b'), (3, 4.0, 5, 'c');
+```
+
+To check the query plan for simple select, execute the following:
+
+```sql
+yugabyte=# EXPLAIN SELECT * FROM employees WHERE k1 = 1;
+```
+
+The following output displays the query execution cost estimate:
+
+```
+QUERY PLAN
+----------------------------------------------------------------
+Foreign Scan on employees  (cost=0.00..112.50 rows=1000 width=44)
+(1 row)
+```
+
+To check the execution plan for select with a complex condition that requires filtering, execute the following:
+
+```sql
+yugabyte=# EXPLAIN SELECT * FROM employees WHERE k1 = 2 and floor(k2 + 1.5) = v1;
+```
+
+The following output displays the cost estimate based on the filtered result:
+
+```
+QUERY PLAN
+----------------------------------------------------------------
+Foreign Scan on employees  (cost=0.00..125.00 rows=1000 width=44)
+   Filter: (floor(((k2)::numeric + 1.5)) = (v1)::numeric)
+(2 rows)
+```
+
+By enabling the `ANALYZE` option and wrapping it to preserve data integrity, you can trigger the query execution, as follows:
+
+```sql
+BEGIN;
+yugabyte=# EXPLAIN ANALYZE SELECT * FROM employees WHERE k1 = 2 and floor(k2 + 1.5) = v1;
+ROLLBACK;
+```
```

Comment:
It might be useful to demonstrate how `EXPLAIN ANALYZE` will not only show the original cost estimates from the query planner, but it will also show what was actually observed by the server during execution.

An example from `EXPLAIN`:
```
yugabyte=# explain select * from foo a left join lateral (select * from foo b where a.a = a.a) c on true;
                              QUERY PLAN
----------------------------------------------------------------------
 Nested Loop Left Join  (cost=0.00..15202.50 rows=5000 width=8)
   Join Filter: (a.a = a.a)
   ->  Seq Scan on foo a  (cost=0.00..100.00 rows=1000 width=4)
   ->  Materialize  (cost=0.00..105.00 rows=1000 width=4)
         ->  Seq Scan on foo b  (cost=0.00..100.00 rows=1000 width=4)
(5 rows)
```

When `EXPLAIN ANALYZE` runs, you get additional information. In this case, we can see how many rescans each node performed (loops), how many milliseconds elapsed before the first row was returned, and total time before the last tuple was returned by each execution node.

```
yugabyte=# explain analyze select * from foo a left join lateral (select * from foo b where a.a = a.a) c on true;
                                                    QUERY PLAN
------------------------------------------------------------------------------------------------------------------
 Nested Loop Left Join  (cost=0.00..15202.50 rows=5000 width=8) (actual time=1.997..8.458 rows=10000 loops=1)
   Join Filter: (a.a = a.a)
   ->  Seq Scan on foo a  (cost=0.00..100.00 rows=1000 width=4) (actual time=1.193..2.501 rows=100 loops=1)
   ->  Materialize  (cost=0.00..105.00 rows=1000 width=4) (actual time=0.008..0.049 rows=100 loops=100)
         ->  Seq Scan on foo b  (cost=0.00..100.00 rows=1000 width=4) (actual time=0.784..4.498 rows=100 loops=1)
 Planning Time: 0.073 ms
 Execution Time: 8.862 ms
(7 rows)
```

Reply:
Added this information as:

In addition to the cost estimates from the query planner, `EXPLAIN ANALYZE` displays the server output produced during the statement execution, as shown in the following example:

```sql
yugabyte=# EXPLAIN ANALYZE SELECT * FROM employees a left LEFT JOIN LATERAL (SELECT * FROM employees b WHERE a.a = b.a) c ON TRUE;
                                 QUERY PLAN
-------------------------------------------------------------------------
 Hash Left Join (cost=112.50..390.00 rows=5000 width=8) (actual time=3.939..6.195 rows=100 loops=1)
   Hash Cond: (a.a = b.a)
   ->  Seq Scan on employees a (cost=0.00..100.00 rows=1000 width=4) (actual time=0.568..2.798 rows=100 loops=1)
   ->  Hash  (cost=100.00..100.00 rows=1000 width=4) (actual time=3.363..3.363 rows=100 loops=1)
         Buckets: 1024  Batches: 1  Memory Usage: 12kB
         ->  Seq Scan on employees b (cost=0.00..100.00 rows=1000 width=4) (actual time=0.458..3.339 rows=100 loops=1)
 Planning Time: 0.939 ms
 Execution Time: 7.089 ms
(8 rows)
```

The server output from the preceding example includes the number of rescans (loops) each node performed, the number of milliseconds passed before the first row was returned, total time before the last tuple was returned by each execution node, and the number of tuples returned by each execution node.

`EXPLAIN`, on the other hand,  does not provide this additional information, as shown in the following examples:

```sql
yugabyte=# EXPLAIN SELECT * FROM employees a LEFT JOIN LATERAL (SELECT * FROM employees b WHERE a.a = b.a) c ON TRUE;
                              QUERY PLAN
----------------------------------------------------------------------
 Hash Left Join (cost=112.50..390.00 rows=5000 width=8)
   Hash Cond: (a.a = b.a)
   ->  Seq Scan on employees a (cost=0.00..100.00 rows=1000 width=4)
   ->  Hash (cost=100.00..100.00 rows=1000 width=4)
         ->  Seq Scan on employees b (cost=0.00..100.00 rows=1000 width=4)
(5 rows)
```